pkg/depsdev/v3alpha/api_test.go

```diff
@@ -522,3 +524,153 @@ func TestGetRequirements(t *testing.T) {
+				Version:        "1.5.0",
+			},
+			{
+				PackageManager: "NPM",
+				PackageName:    "defangjs",
+				Version:        "1.0.7",
+			},
+		})
+
+		require.Nil(t, err)
+		assert.NotNil(t, iter)
+
+		defer iter.Close()
+
+		expected := []def.Version{
+			{
+				VersionKey: def.VersionKey{
+					System:  "NPM",
+					Name:    "@colors/colors",
+					Version: "1.5.0",
+				},
+				IsDefault:    false,
+				Licenses:     []string{"MIT"},
+				AdvisoryKeys: []def.AdvisoryKeys{},
+				Links: []def.Links{
+					{
+						Label: "HOMEPAGE",
+						URL:   "https://github.com/DABH/colors.js"},
+					{
+						Label: "ISSUE_TRACKER",
+						URL:   "https://github.com/DABH/colors.js/issues",
+					},
+					{
+						Label: "ORIGIN",
+						URL:   "https://registry.npmjs.org/@colors%2Fcolors/1.5.0",
+					},
+					{
+						Label: "SOURCE_REPO",
+						URL:   "git+ssh://git@github.com/DABH/colors.js.git",
+					},
+				},
+				SlsaProvenances: []def.SLSAProvenances{},
+				PublishedAt:     time.Date(2022, time.February, 12, 7, 39, 4, 0, time.UTC),
+				Registries:      []string{"https://registry.npmjs.org/"},
+				RelatedProjects: []def.RelatedProjects{
+					{
+						ProjectKey:         def.ProjectKey{ID: "github.com/dabh/colors.js"},
+						RelationProvenance: "UNVERIFIED_METADATA",
+						RelationType:       "ISSUE_TRACKER",
+					},
+					{
+						ProjectKey:         def.ProjectKey{ID: "github.com/dabh/colors.js"},
+						RelationProvenance: "UNVERIFIED_METADATA",
+						RelationType:       "SOURCE_REPO"},
+				},
+			},
+			{
+				VersionKey: def.VersionKey{
+					System:  "NPM",
+					Name:    "defangjs",
+					Version: "1.0.7",
+				},
+				IsDefault:    true,
+				Licenses:     []string{"GPL-3.0"},
+				AdvisoryKeys: []def.AdvisoryKeys{},
+				Links: []def.Links{
+					{
+						Label: "HOMEPAGE",
+						URL:   "https://github.com/edoardottt/defangjs#readme",
+					},
+					{
+
+						Label: "ISSUE_TRACKER",
+						URL:   "https://github.com/edoardottt/defangjs/issues",
+					},
+					{
+
+						Label: "ORIGIN",
+						URL:   "https://registry.npmjs.org/defangjs/1.0.7",
+					},
+					{
+
+						Label: "SOURCE_REPO",
+						URL:   "git+https://github.com/edoardottt/defangjs.git",
+					},
+				},
+				SlsaProvenances: []def.SLSAProvenances{},
+				PublishedAt:     time.Date(2023, time.May, 16, 9, 48, 31, 0, time.UTC),
+				Registries:      []string{"https://registry.npmjs.org/"},
+				RelatedProjects: []def.RelatedProjects{
+					{
+						ProjectKey:         def.ProjectKey{ID: "github.com/edoardottt/defangjs"},
+						RelationProvenance: "UNVERIFIED_METADATA",
+						RelationType:       "ISSUE_TRACKER",
+					},
+					{
+						ProjectKey:         def.ProjectKey{ID: "github.com/edoardottt/defangjs"},
+						RelationProvenance: "UNVERIFIED_METADATA",
+						RelationType:       "SOURCE_REPO",
+					},
+				},
+			},
+		}
+		results, err := consumeIter(iter)
+		require.NoError(t, err)
+
+		assert.Equal(t, expected, results)
```

Comment:
Like previously in other discussions, I don't know if this is the best way to check the correctness of code.
E.g. I could release a new version of defangjs and this test case will fail :/

Reply:
No it will not since we're requesting details of a specific version. The logic is the same used in the `TestGetVersion` test.